include/openssl/evp.h

```diff
@@ -181,6 +181,9 @@ int (*EVP_MD_meth_get_ctrl(const EVP_MD *md))(EVP_MD_CTX *ctx, int cmd,
  */
 # define EVP_MD_CTX_FLAG_FINALISE        0x0200
 
+/* Don't free up ctx->pctx in EVP_MD_CTX_reset */
+# define EVP_MD_CTX_FLAG_NEGLECT_PCTX    0x0400
```

Comment:
This feels misplaced in the public header - although I understand why you put it there.

Reply:
It should also have a different name - it is do-not-free and neglect isn't the right name to use.
Perhaps EVP_MD_CTX_FLAG_KEEP_PKEY_CTX ... I'd also not abbreviate to PCTX ...